packages/api_bo/src/resolvers/query.ts

```diff
@@ -1,5 +1,36 @@
+import { ObjectId } from "objectId";
+import { Context } from "../app.ts";
+import { centerCollection, CenterModel } from "../models/CenterModel.ts";
+import { groupCollection } from "../models/GroupModel.ts";
+import { QueryGetCenterArgs } from "../types.ts";
+
 export const Query = {
-  hello: (): string => {
-    return "hello world";
+  getCenters: async (_parent: unknown, _args: unknown, ctx: Context) => {
+    return await centerCollection(ctx.db).find().toArray();
+  },
+
+  getCenter: async (
+    _parent: unknown,
+    args: QueryGetCenterArgs,
+    ctx: Context,
+  ) => {
+    const center = await centerCollection(ctx.db).findOne({
+      _id: new ObjectId(args.id),
+    });
+    if (!center) {
+      throw new Error("Center not found");
+    }
+    return center;
+  },
+};
+
+export const Center = {
+  id: (parent: CenterModel): string => {
+    return String(parent._id!);
+  },
+  groups: async (parent: CenterModel, ctx: Context) => {
+    return await Promise.all(parent.groups.map(async (group: ObjectId) => {
+      return await groupCollection(ctx.db).findOne({ _id: group });
```

Comment:
No entiendo muy bien qué quieres hacer aquí... parent.groups es un array de ObjectId? No se puede hacer con un solo find directamente
`groupCollection(ctx.db).find({ _id: {$in: parent.groups} })` ?

Reply:
esto lo intente y no cogía nada

Reply:
mmm igual por el tipo? casteando a objectId tampoco?  `{$in: parent.groups.map(group => new ObjectId(group))}`
Es que hacer n llamadas de mongo cuando se puede resolver en una no es muy eficiente... :sweat_smile:

Reply:
Jajajajaja toda la razón lo intento con eso a ver si así funciona

Reply:
lo he intentando casteando y me d el mismo error: "message": "Cannot read properties of undefined (reading 'collection')"